src/core/protocol-authorization.ts

```diff
@@ -283,62 +261,59 @@ export class ProtocolAuthorization {
    * Verifies the actions specified in the given message matches the allowed actions in the rule set.
    * @throws {Error} if action not allowed.
    */
-  private static verifyAllowedActions(
+  private static async verifyAllowedActions(
```

Comment:
This method here is the centerpoint of the PR. Looking for feedback here and in functions called herein.